src/utils/sources-tree.js

```diff
@@ -73,14 +73,31 @@ function createParentMap(tree: any): WeakMap<any, any> {
  * @memberof utils/sources-tree
  * @static
  */
+function getFilenameFromPath(pathname) {
+  let filename = "";
+  if (pathname) {
+    filename = pathname.substring(pathname.lastIndexOf("/") + 1);
+    // This file does not have a name. Default should be (index).
+    if (filename == "" || !filename.includes(".")) {
+      filename = "(index)";
+    }
+  }
+  return filename;
+}
+
+/**
+ * @memberof utils/sources-tree
+ * @static
+ */
 function getURL(sourceUrl: string): { path: string, group: string } {
   const url = sourceUrl;
-  let def = { path: "", group: "" };
+  let def = { path: "", group: "", filename: "" };
   if (!url) {
     return def;
   }
 
   const { pathname, protocol, host, path } = parse(url);
+  const filename = getFilenameFromPath(pathname);
```

Comment:
flow is complaining about this line.  I believe `pathname` could be a string or also undefined and the function only accepts a string.  you could check `pathname` and just set filename to "" if pathname undefined.